controllers/spec/common.go

```diff
@@ -884,7 +903,11 @@ func getProcessPythonRuntimeArgs(name, packageName, clusterName, details, uid st
 
 // This method is suitable for Java and Python runtime, not include Go runtime.
 func getSharedArgs(details, clusterName, uid string, authProvided bool, tlsProvided bool,
-	tlsConfig TLSConfig, authConfig *v1alpha1.AuthConfig) []string {
+	tlsConfig TLSConfig, authConfig *v1alpha1.AuthConfig, healthCheckInterval *int32) []string {
+	var hInterval int32 = -1
+	if healthCheckInterval != nil && *healthCheckInterval > 0 {
+		hInterval = *healthCheckInterval
```

Comment:
how can the function-mesh controller know whether a runner image has the `grpcurl`?

Reply:
Or is it possible to make `grpcurl` can be used in released images?

Reply:
yes, that's the plan, so the default runner images support health check feature, and for custom runner image, users' need to make sure it work with it

Reply:
but for builtin io-connectors, we need to update them one-by-one

Reply:
just an idea, how about sharing a volume of a grpcurl image to make the runner image have the grpcurl executable as well?

Reply:
hm, or add the grpcurl and .proto downloading commands to the runner's command?

Reply:
adding more commands to the runner may make the distroless way even harder (sad)

Reply:
ok, then we need to maintain a custom grpcurl image that contains the `InstanceCommunication.proto`, and share an volume with runner containers

Reply:
we can add a field: `spec.liveness. persistentVolumeClaim`, if it's set, we assume it contains what we need, and create Volume&Volume Mounts for liveness probe automatically

Reply:
only `spec.liveness. persistentVolumeClaim` may not be enough since we have to `exec` the actual executable. so such configs would be better if it is managed by function-mesh controller.

something like:
- function mesh creates a PVC when installed to the kubernetes cluster
- run a `Job` to move the contents from an image to the volume
- mount the volume to each Function/Connector if `healthCheckInterval` is enabled, and apply the liveness probe as well

Reply:
sounds good, I will update

Reply:
I am not very familiar with PVC usage in different cloud vendors, so not sure if the above steps are reasonable to go production. @nlu90 @tpiperatgod PTAL when you have time, thanks.